client/src/components/Island/index.tsx

```diff
@@ -176,55 +171,37 @@ const IslandContainer = styled.div<{ minWidth }>`
 const IslandImage = styled.img<{ maxWidth }>`
   max-width: ${(props) => `${props.maxWidth}px`};
   max-height: ${(props) => `${props.maxWidth}px`};
-`;
-
-const IslandToolTipContainer = styled.div`
   position: absolute;
   top: 0;
   left: 0;
 `;
 
-const ToolTipSquare = styled.div<{ width }>`
-  width: ${(props) => `${props.width}px`};
-  height: ${(props) => `${props.width}px`};
-  z-index: 9;
-  position: relative;
-  display: flex;
-  flex-wrap: wrap;
-`;
-
-function getIslandToolTipText(classObject: any) {
-  return (
-    <>
-      <MarginDiv>Name: {classObject.name}</MarginDiv>
-      <MarginDiv>Type: {classObject.type}</MarginDiv>
-      <MarginDiv>Access modifier: {classObject.accessModifier}</MarginDiv>
-      <MarginDiv>Lines: {classObject.lineCount}</MarginDiv>
-    </>
-  );
-}
-
 const Island: React.FC = () => {
-  const toolTipArray = getToolTipDataArray(mockClass);
-  const numberOfIcons = toolTipArray.length;
-  const square = Math.sqrt(numberOfIcons);
-  const numberHorizontal = Math.ceil(square);
-  const width = numberHorizontal * iconWidth;
+  const toolTipArray = mockClasses.map((c) => getToolTipDataArray(c));
+  const [width, setWidth] = useState(0);
+  const minIslandWidth = width + iconWidth * 2;
 
-  const minimumInnerIslandWidth = width + iconWidth * 2;
   const numberOfIslandImages = islandArray.length;
   const randomIslandIndex = Math.floor(Math.random() * numberOfIslandImages);
   const islandImage = islandArray[randomIslandIndex];
 
-  const islandToolTipText = getIslandToolTipText(mockClass);
+  let numberOfLines = 0;
+  mockClasses.forEach((c) => {
+    numberOfLines = numberOfLines + c.lineCount;
+  });
+  // TODO 1px = 1 line is arbitrary, adjust as desired.
+  const fileSizeAdjustedWidth =
+    minIslandWidth > numberOfLines ? minIslandWidth : numberOfLines;
+
```

Comment:
Do we also need to set a maximum width?

Reply:
good catch! I've set the maximum island width based on line count to 400px for now (which is pretty huge), but maybe it would be good to look into logarithmic increases